File: src/csharp/api/Models/Patient.cs
using Example.Api.Enums;

namespace Example.Api.Models;

/// <summary>
/// Represents a patient in the system.
/// </summary>
public record Patient : BaseEntity
{
    /// <summary>
    /// Gets or sets the unique identifier for the patient.
    /// </summary>
    public long Id { get; set; }

    /// <summary>
    /// Gets or sets the age of the patient.
    /// </summary>
    /// <value></value>
    public int Age { get; set; }

    /// <summary>
    /// Gets or sets the gender of the patient.
    /// </summary>
    /// <value></value>
    public Gender Gender { get; set; }

    /// <summary>
    /// Get or sets the email of the patient.
    /// </summary>
    /// <value></value>
    public string Email { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the phone number of the patient.
    /// </summary>
    /// <value></value>
    public string PhoneNumber { get; set; } = string.Empty;

    /// <summary>
    /// Gets or sets the address of the patient.
    /// </summary>
    /// <value></value>
    public Address Address { get; set; } =
        new Address(string.Empty, string.Empty, string.Empty, string.Empty, string.Empty, string.Empty);

    /// <summary>
    /// Gets or sets the full name of the patient.
    /// </summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>
    /// Get or sets the date of birth of the patient.
    /// </summary>
    /// <value></value>
    public DateTimeOffset DateOfBirth { get; set; }

    /// <summary>
    /// Gets or sets the first visit date of the patient.
    /// </summary>
    /// <value></value>
    public DateTimeOffset FirstVisitDate { get; set; }

    /// <summary>
    /// Get or sets the status of the patient.
    /// </summary>
    /// <value></value>
    public PatientStatus Status { get; set; }

    /// <summary>
    /// Get or sets any additional remarks about the patient.
    /// </summary>
    /// <value></value>
    public string? Remarks { get; set; }

    /// <summary>
    /// Gets or sets the collection of orders associated with the patient.
    /// </summary>
    public ICollection<PatientOrder> Orders { get; set; } = [];
}

/// <summary>
/// Pyhsical address representation.
/// </summary>
/// <param name="Country">Country</param>
/// <param name="City">City</param>
/// <param name="Area">Area</param>
/// <param name="Road">Road</param>
/// <param name="Street">Street</param>
/// <param name="Others">Others</param>
/// <returns></returns>
public record Address(string Country, string City, string Area, string Road, string Street, string Others);

File: src/python/app/entities/patient.py
"""Patient entity model."""

from __future__ import annotations
from typing import List, TYPE_CHECKING, Optional
from datetime import date, datetime

from app.entities.base import Base
from app.entities.base_entity import BaseEntity
from app.enums.patient import Gender, PatientStatus
from app.db.types import IntEnumType
from sqlalchemy import String, Integer, Date, DateTime, Index, text, BigInteger
from sqlalchemy.orm import Mapped, mapped_column, relationship

if TYPE_CHECKING:
    from app.entities.patient_order import PatientOrder


class Patient(Base, BaseEntity):
    """Patient model."""

    __tablename__ = "patient"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    name: Mapped[str] = mapped_column(String(50), nullable=False)
    age: Mapped[int] = mapped_column(Integer, nullable=False)
    gender: Mapped[Gender] = mapped_column(IntEnumType(Gender), nullable=False)
    email: Mapped[Optional[str]] = mapped_column(String(100), nullable=True)
    phone_number: Mapped[str] = mapped_column(String(10), nullable=False)

    # Address owned fields (flattened) matching EF Core names
    country: Mapped[Optional[str]] = mapped_column(String(25), nullable=True)
    city: Mapped[Optional[str]] = mapped_column(String(25), nullable=True)
    area: Mapped[Optional[str]] = mapped_column(String(25), nullable=True)
    road: Mapped[Optional[str]] = mapped_column(String(25), nullable=True)
    street: Mapped[Optional[str]] = mapped_column(String(25), nullable=True)
    address_others: Mapped[Optional[str]] = mapped_column(String(100), nullable=True)

    date_of_birth: Mapped[date] = mapped_column(Date, nullable=False)
    first_visit_date: Mapped[datetime] = mapped_column(DateTime(timezone=True), nullable=False)
    status: Mapped[PatientStatus] = mapped_column(
        IntEnumType(PatientStatus), nullable=False, default=PatientStatus.Active
    )
    remarks: Mapped[Optional[str]] = mapped_column(String(500), nullable=True)

    # Relationship
    orders: Mapped[List["PatientOrder"]] = relationship(
        "PatientOrder", back_populates="patient", cascade="all, delete-orphan", order_by="desc(PatientOrder.id)"
    )
